### PR TITLE
musl: new, 1.2.3

### PR DESCRIPTION
**Pull request by dynos01**
Topic Description
-----------------
musl: new, 1.2.3

Package(s) Affected
-------------------
musl: new, 1.2.3

Security Update?
----------------
No

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

**Comment by dynos01** [comment]
musl

File: extra-devel/musl/autobuild/build [hunk 1]
```diff
@@ -0,0 +1,18 @@
+abinfo "Building musl... "
+
+./configure --prefix=/usr/lib/musl \
```
Comment (MingcongBai) [inline extra-devel/musl/autobuild/build]: Unless this is not a standard GNU Autotools source, please use `ABTYPE=autotools` and rename this file to `beyond`, cut everything before line 14.

File: extra-devel/musl/autobuild/build [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    --enable-wrapper=all \
+    CFLAGS="-ffat-lto-objects"
+
+make
+
+abinfo "Installing musl... "
+
+DESTDIR="$PKGDIR" make install
+
+mv "$PKGDIR"/lib/ld-musl*.so* "$PKGDIR"/usr/lib/
```
Comment (MingcongBai) [inline extra-devel/musl/autobuild/build]: This would have been handled by the ABTYPE.

File: extra-devel/musl/autobuild/build [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+rmdir "$PKGDIR"/lib
+
+install -Dvm644 README "$PKGDIR"/usr/share/doc/musl/README
```
Comment (MingcongBai) [inline extra-devel/musl/autobuild/build]: Autobuild3 handles all copyright/license/README-type files automatically.

File: extra-devel/musl/autobuild/defines [hunk 4]
```diff
@@ -0,0 +1,7 @@
+PKGNAME=musl
+PKGSEC=devel
+BUILDDEP="devel-base"
```
Comment (MingcongBai) [inline extra-devel/musl/autobuild/defines]: No need to specify this.

File: extra-devel/musl/autobuild/defines [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+PKGDES="Lightweight implementation of C standard library"
+
+NOSTATIC=0
+ABSPLITDBG=0
```
Comment (MingcongBai) [inline extra-devel/musl/autobuild/defines]: Why?

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-devel/musl/autobuild/build (hunk 1 above)
Comment (dynos01) [reply extra-devel/musl/autobuild/build]: This is not a standard GNU Autotools source, so `ABTYPE=autotools` won't work.

File: extra-devel/musl/autobuild/build (hunk 3 above)
Comment (dynos01) [reply extra-devel/musl/autobuild/build]: Remove these lines.

File: extra-devel/musl/autobuild/defines (hunk 4 above)
Comment (dynos01) [reply extra-devel/musl/autobuild/defines]: Removed `BUILDDEP="devel-base"`.

File: extra-devel/musl/autobuild/defines (hunk 5 above)
Comment (dynos01) [reply extra-devel/musl/autobuild/defines]: There is no `BuildID[sha1]` for several `.o` files, so Autobuild3 will fail without `ABSPLITDBG=0`.

File: extra-devel/musl/autobuild/build (hunk 1 above)
Comment (MingcongBai) [reply extra-devel/musl/autobuild/build]: Okay, then please use absolute path for `configure` - `"$SRCDIR"/configure`.

File: extra-devel/musl/autobuild/defines (hunk 5 above)
Comment (MingcongBai) [reply extra-devel/musl/autobuild/defines]: Are there any shared objects?

File: extra-devel/musl/autobuild/build (hunk 2 above)
Comment (MingcongBai) [reply extra-devel/musl/autobuild/build]: Add verbosity to mv, `mv -v`.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-devel/musl/autobuild/build (hunk 1 above)
Comment (dynos01) [reply extra-devel/musl/autobuild/build]: Done.

File: extra-devel/musl/autobuild/build (hunk 2 above)
Comment (dynos01) [reply extra-devel/musl/autobuild/build]: Done.

File: extra-devel/musl/autobuild/defines (hunk 5 above)
Comment (dynos01) [reply extra-devel/musl/autobuild/defines]: musl-libc
musl provides libc.so.

Comment (MingcongBai) [reply extra-devel/musl/autobuild/defines]: Then, in this case, you should see if flags defined by Autobuild3 were passed into the build system.

File: extra-devel/musl/autobuild/build [hunk 6]
```diff
@@ -0,0 +1,15 @@
+abinfo "Building musl... "
+
+"$SRCDIR"/configure --prefix=/usr/lib/musl \
+    --exec-prefix=/usr \
+    --enable-wrapper=all \
+    CFLAGS="-ffat-lto-objects"
```
Comment (MingcongBai) [inline extra-devel/musl/autobuild/build]: Why was this necessary? Add a comment to explain (FIXME marker).

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Review by MingcongBai — APPROVED** [review]
